EosPayload/lib/mqtt/client.py

```diff
@@ -8,20 +8,38 @@
 class Client(mosquitto.Client):
     
     def __init__(self, host: str, port: int = 1883):
+        """ Connects to the MQTT server and spawns a thread for async MQTT operations.
+            Connect operation is synchronous / blocking.  Subsequent sends/receives are async.
+
+        :param host: the hostname of the MQTT server
+        :param port: the port of the MQTT server
+        """
         super(Client, self).__init__(protocol=mosquitto.MQTTv5)
         self.connect(host, port)
-
-    def receive(self) -> int:
-        return self.loop_forever()
+        self.loop_start()
 
     def send(self, topic: Topic, payload: str) -> bool:
+        """ Send an MQTT message.
+            Async (Non-Blocking).
+
+        :param topic: the topic to send
+        :param payload: the stringified message body
+        :return: True on success, False otherwise
+        """
         msg_info = self.publish(topic, payload, QOS.DELIVER_AT_MOST_ONCE)
+        msg_info.wait_for_publish()
         if msg_info.rc != mosquitto.MQTT_ERR_SUCCESS:
             print("MQTT send failed with error code " + msg_info.rc)
             return False
         return True
 
-    def register_subscriber(self, topic: Topic, callback: Callable) -> Any:
-        return self.message_callback_add(topic, callback)
+    def register_subscriber(self, topic: Topic, callback: Callable) -> None:
```

Comment:
I thought about this and the answer is probably not for overhead reasons but there might be a good way to still make implementing a callback as easy as overriding something in driver_base

Reply:
I don't think this makes sense because there is no standard action to take for all messages.  For example, a Driver that wants to listen in on temperature data from another sensor but also take it's own commands.  And I don't think there's much benefit to an overridable function that's just 1 giant switch case